app/controllers/admin/claims_controller.rb

```diff
@@ -22,14 +22,16 @@ def show
   end
 
   def search
-    return unless params[:reference].present?
+    return unless params[:query].present?
 
-    claim = Claim.find_by(reference: params[:reference].upcase)
+    @claims = Claim::Search.new(params[:query]).claims.includes(:eligibility)
 
-    if claim
-      redirect_to(admin_claim_url(claim))
+    if @claims.blank?
+      flash.now[:notice] = "Cannot find a claim for query \"#{params[:query]}\""
+    elsif @claims.count == 1
```

Comment:
Ruby's Enumerable has a `one?` method:

```suggestion
    elsif @claims.one?
```

Reply:
Ooh, nice